src/controllers/JobController.js

```diff
@@ -61,6 +61,9 @@ async function searchJobs (req, res) {
     req.query.jobIds = req.body.jobIds
   }
   const result = await service.searchJobs(req.authUser, req.query)
+  if (req.query.jobIds) {
+    delete req.query.jobIds
+  }
```

Comment:
The current logic to add and then to delete fields is a bit hard to follow. It would be nice if logic would more straightforward, for example like this:

```js
const query = {
   ...req.query,
   // we can pass jobIds in `body` or in `query`
   jobIds: _.get(req, 'body.jobIds', req.query.jobIds)
}
const result = await service.searchJobs(req.authUser, query)

Reply:
I agree, maybe simply check body for it, if jobIds are not present, we can avoid to pass jobIds to the criteria or  pass an empty array.  By looking at the service method, seems both would be fine.